src/targets/node/terminalNodeLauncher.ts

```diff
@@ -174,6 +175,35 @@ export class TerminalNodeLauncher extends NodeLauncherBase<ITerminalLaunchConfig
     });
   }
 
+  private async writeCommand(terminal: vscode.Terminal, command: string) {
+    // Wait for shell integration if we expect it to be available
+    const config = vscode.workspace.getConfiguration();
+    if (!terminal.shellIntegration && config.get(SHELL_INTEGRATION_SETTING) !== false) {
+      let listener: vscode.Disposable;
+      await Promise.race([
+        delay(SHELL_INTEGRATION_TIMEOUT),
```

Comment:
For python we're trying to do some caching mechanism to skip this timeout depending on whether the terminal is likely to have shell integration. You could track whether the shell exe and args had shell integration last time and if not revert to sendText immediately (or after some small timeout if needed to avoid this pwsh problem).

Reply:
Wonder if the API could signal that to avoid all of us duplicating logic. Some `terminal.isLikelyToHaveShellIntegration`

Reply:
I've thought about that, but it's such an ugly API and doubt it would make it through the syncs tbh. There are some basic examples of fallbacks in the comments but @anthonykim1/Python was about to try out this idea soon.

Reply:
It doesn't look like the shell exe / args are available on the vscode.Terminal object, do you have suggestions on how to get those?

Reply:
They're in creationOptions, if the shell path is undefined you can assume it's env.shell